update maintenance guide with recent news (#419)
rust-lang/flate2-rs: MAINTENANCE.md

This document explains how to perform the project's maintenance tasks.

### Repository Configuration

Settings can be changed here by means of a PR - depending on what it is a conversation about pro's and cons might be needed.

https://github.com/rust-lang/team/blob/master/repos/rust-lang/flate2-rs.toml

You can send a PR to add `required-approvals = 0` to the configuration file above. See [this configuration](https://github.com/rust-lang/team/blob/master/repos/rust-lang/libc.toml) 
for an example.

### Creating a new release

#### Artifacts

* a tag of the version number
* a new [crate version](https://crates.io/crates/flate2/versions)

> #### Process: Plain `cargo publish`

Reviewer: I would like an introductory paragraph to describe why there are two release processes, and which one to use by default.
```suggestion
#### Process
We are moving to using `cargo smart-release` for releases, but it still needs some polishing. [Some criteria to choose which one to use.]
##### `cargo publish`
...
##### `cargo smart-release`
...
```

Author: Actually, this could be up to discussion if having two processes even makes sense.
One day I just 'felt' very strongly about wanting to use `cargo smart-release` but didn't think of the incompatibility regarding the tags, and the other problems that might arise from a PR-based workflow which `cargo smart-release` isn't designed around (so `gh` invocations might fail).

Even though Changelogs are nice, in the end, it might be easier to remove it and call it a failed experiment - I'd actually vote for that.

Reviewer: Reverting the changes from the Process header down would be clearer. For changelogs, is it possible to add `cargo changelog` to the `cargo publish` workflow?

Author: That’s definitely possible.

6. `git push --tags` to push the new tag.
7. Go to the newly created release page on GitHub and edit it by pressing the "Generate Release Notes" and the `@` button. Save the release.

#### Alternative Process: `cargo smart-release`

This also maintains the `changelog.md` file, and it's able to pick up previous `cargo publish` releases and thus mixes well.
However, it's quite new and needs some polishing.

To just update the changelog, run `cargo changelog --write`.

To create a new release, do as follows:

1. `git checkout -b release-<next-version>` - move to a branch to prepare making changes to the repository. *Changes cannot be made to `main` as it is protected.*
2. `cargo changelog --write` to update the changelog. Edit it to your liking.
3. Create a commit like `git commit -am "update changelog prior to release"`

> 4. `cargo smart-release -e --update-crates-index` - double-check that the release version is sensible, and if not, just edit `Cargo.toml` by hand before.

Reviewer: Does this create a second commit or update the commit from step 3?

> 4. `cargo smart-release -e --update-crates-index` - double-check that the release version is sensible, and if not, just edit `Cargo.toml` by hand before.

Reviewer: ```suggestion
4. `cargo smart-release -e --update-crates-index` - double-check that the release version is sensible, and if not, edit `Cargo.toml` to specify a sensible release.
```

> 5. Tags were created, and they should be pushed, but it might be that it fails to create a GitHub release. 

Reviewer: Is this due to use of `v...` tag? Any way to configure `smart-release` to use tags without the `v`?

6. Create an additional release tag with the `v` removed, i.e. `git tag -s <version> v<version>`. Then `git push  --tags` by hand.

> 7. Double-check the newly created release on GitHub is populated - if not, generate the changelog manually.

Reviewer: It looks like the release for 1.0.31 was not populated. Can you describe the steps more clearly here, perhaps by doing them for 1.0.31?
